ngraph/core/src/function.cpp

```diff
@@ -520,3 +531,121 @@ ov::op::util::Variable::Ptr ov::Function::get_variable_by_id(const string& varia
 }
 
 constexpr ov::DiscreteTypeInfo ov::AttributeAdapter<shared_ptr<ov::Function>>::type_info;
+
+/// Output functions
+std::vector<ov::Output<const ov::Node>> ov::Function::outputs() const {
+    std::vector<ov::Output<const ov::Node>> results;
+    for (const auto& res : m_results) {
+        std::shared_ptr<const ov::Node> result = res;
+        results.emplace_back(result);
+    }
+    return results;
+}
+ov::Output<const ov::Node> ov::Function::output() const {
+    if (m_results.size() != 1) {
+        throw ov::Exception("output() must be called on a function with exactly one result.");
+    }
+    std::shared_ptr<const ov::Node> result = m_results.at(0);
+    return result;
+}
+ov::Output<const ov::Node> ov::Function::output(size_t i) const {
+    std::shared_ptr<const ov::Node> result = m_results.at(0);
```

Comment:
Shouldn't it be `m_results.at(i)`?

Reply:
looks like the tests don't catch it because it's called only for const ov::Function

Reply:
Fixed